src/InputOutput/IOstrings/IOstrings_outpath_name.jl

```diff
@@ -0,0 +1,42 @@
+#
+# Define output path string:
+#
+# called by a driver right before run()
+#
+# ex. 
+#    #Create unique output path directory:
+#    OUTPATH = IOstrings_outpath_name(problem_name, grid_resolution)
+#
+#
+function IOstrings_outpath_name(problem_name, grid_resolution)
+    #
+    # Arguments:
+    # problem_name = "NAME"
+    # grid_resolution[1]   = Δx
+    # grid_resolution[2]   = Δy
+    # grid_resolution[end] = Δz
+    #
+    ndim = length(grid_resolution)
+
+    outpath_string = string(grid_resolution[1], "mx")
+    for i = 2:ndim
+        ds = grid_resolution[i]
+        outpath_string = string(outpath_string, ds, "mx")
+    end        
+    OUTPATH = string("./output/",problem_name, "/", outpath_string,"_", randstring(6))
+
+    #                                                                                                                                                                                                                                          
+    # Create output directories to store diagnostics and vtks:
+    #
+    if (isdir(OUTPATH) == false)
+        mkpath(OUTPATH)
+    else
+        #Move an existing directory to dir_previous if it already exists
+        auxi = OUTPATH[1:end]
+        previous_run = string(OUTPATH, "_previous");
+        mv(OUTPATH, previous_run, force=true)
```

Comment:
Overall, I love the idea of this part of your PR (making a dedicated output directory tree). However, this looks a bit dangerous to me, it will overwrite previous runs if the names are the same, which means you need to keep track of it. I think we need to think of a more elegant solution here.

Reply:
Hi @charleskawczynski this became obsolete after my last push and forgot to remove it (I will do that now). I am appending a random hash to the name of every new directory that is created to make them unique. This said, I will also replace this solution by appending the current timestamp.